### PR TITLE
Remove boost deprecations

### PR DESCRIPTION
**Pull request by Dirk007**
Fixes:
- boost/test/floating_point_comparison.hpp:14:124: note: #pragma message: This header is deprecated. Use This header is deprecated. Please use <boost/test/tools/floating_point_comparison.hpp> instead. instead.
- boost/detail/sp_typeinfo.hpp:23:54: note: #pragma message: This header is deprecated. Use <boost/core/typeinfo.hpp> instead.

### CODE REVIEW

File: include/turtle/constraints.hpp [hunk 1]
```diff
@@ -19,7 +19,7 @@
 #include <boost/type_traits/common_type.hpp>
 #include <boost/type_traits/is_convertible.hpp>
 #include <boost/type_traits/has_equal_to.hpp>
-#include <boost/test/floating_point_comparison.hpp>
+#include <boost/test/tools/floating_point_comparison.hpp>
```
Comment (mat007) [inline include/turtle/constraints.hpp]: Does this not need a `#if BOOST_VERSION`?

File: include/turtle/detail/type_name.hpp [hunk 2]
```diff
@@ -24,7 +28,7 @@
 #include <cstdlib>
 #endif
 
-#define MOCK_TYPE_NAME( t ) mock::detail::type_name( BOOST_SP_TYPEID(t) )
+#define MOCK_TYPE_NAME( t ) mock::detail::type_name( BOOST_CORE_TYPEID(t) )
```
Comment (mat007) [inline include/turtle/detail/type_name.hpp]: Doesn’t this depend on which include was used?
I think this should be wrapped in a macro such as `MOCK_TYPEID`?

File: include/turtle/detail/type_name.hpp [hunk 3]
```diff
@@ -33,7 +37,7 @@ namespace detail
     class type_name
     {
     public:
-        explicit type_name( const boost::detail::sp_typeinfo& info )
+        explicit type_name( const boost::core::typeinfo& info )
```
Comment (mat007) [inline include/turtle/detail/type_name.hpp]: Same here it probably depends on whether we included `boost/core` or `boost/detail`?

File: include/turtle/object.hpp [hunk 4]
```diff
@@ -51,7 +51,7 @@ namespace detail
             : impl_( boost::make_shared< detail::object_impl >() )
         {}
     protected:
-        ~object()
+        virtual ~object()
```
Comment (mat007) [inline include/turtle/object.hpp]: Can you explain why this change?

File: include/turtle/detail/type_name.hpp (hunk 2 above)
Comment (Dirk007) [reply include/turtle/detail/type_name.hpp]: For 1.66 the definition was `#define BOOST_SP_TYPEID(T) BOOST_CORE_TYPEID(T)` - thus it should not make any difference. Can not say for older versions though.

File: include/turtle/constraints.hpp (hunk 1 above)
Comment (Dirk007) [reply include/turtle/constraints.hpp]: Good question indeed. It does not harm if we add it.. will do

File: include/turtle/detail/type_name.hpp (hunk 3 above)
Comment (Dirk007) [reply include/turtle/detail/type_name.hpp]: Probably we could also just throw a `#if BOOST_VERSION` into the place here.. what do you think?

File: include/turtle/object.hpp (hunk 4 above)
Comment (Dirk007) [reply include/turtle/object.hpp]: Oh yeah, this accidently made it to here. It is the issue from https://github.com/mat007/turtle/pull/44 - that PR came from a former colleague. Unfortunately I did not have the time to step deep into the topic, so I just made the ctor virtual to get our socket mock to work again 🤷‍♂

File: include/turtle/detail/type_name.hpp (hunk 2 above)
Comment (Dirk007) [reply include/turtle/detail/type_name.hpp]: Just took a quick look at 1.54 - u r right. Will change.

**Comment by Dirk007** [comment]
I somehow managed to f*** up my branch whilst cleaning up. As I am in a big hurry for the last working day this year, I just opened a new one here https://github.com/mat007/turtle/pull/80
sorry :D
